Sprint 5

O-o-l-e-g opened the pull request.

KonstantinChikishevWG committed:
- Fixing lenta controller

grigorii-sum committed:
Добавлен endpoint для отправки сообщений в Kafka
Изменена структура пакета "parser" (добавлены интерфейсы и в них вынесено описание классов и методов)
Добавлены логи

grigorii-sum committed:
Merge branch 'sprint-5' of https://github.com/microservices-course-itmo/lenta-parser-service into sprint-5

# Conflicts:
#	src/main/java/com/wine/to/up/lenta/service/controller/LentaStoreController.java
#	src/main/java/com/wine/to/up/lenta/service/parser/ParserRsp.java

grigorii-sum committed:
refactoring

butenko-or committed:
Change for checking deploy

KonstantinChikishevWG committed:
- Replacing UpdatingProducts to ParserApi

viprikz committed:
- Adding Tests

kosch71 committed:
- Fixing job
- Fixing kafka controller
- Refactoring ProductDTO
- Tiny fix in EventSerializer
- Wait while Eugeniy complete parser, then i will LentaWineParserServiceImpl

kosch71 committed:
- Adding checking in ExportProductDtoList and SendKafkaController while we wait completed parser
- Deleting some build properties in LentaWineParserServiceImpl
- Changing LentaStoreController Response

kosch71 committed:
- Move Tests

kosch71 committed:
- Adding Selenium in pom.xml
- Refactoring LentaWineParserServiceImpl(Adding new properties)
- Adding new property in lenta-site.properties

eomelyanovich committed:
Изменение парсера: парсинг новых пропертей и драйвер для селениума

grigorii-sum committed:
Tests

grigorii-sum committed:
Merge remote-tracking branch 'origin/sprint-5' into sprint-5

eomelyanovich committed:
Refactoring parser: full parsing, changing parser config

kosch71 committed:
## Parser rsp integration
- Adding normal unpacking of json
- Adding DRY sugar and RED colour as default because unrecognized throw Ex.(common-parser-api)
- Refactoring properties in lenta-site.properties
- Uncommenting packing from our dto to api dto

viprikz committed:
Parser test

kosch71 committed:
## Refactoring from Crossroad team
- Adding ControllerConfiguration
- Adding ExportProductDtoList in SendKafkaController
- Adding try/catch blocks in ParserReqServiceImpl
- Adding ColourTest, SugarTest
- Removing selenium support and adding jsoup parsing in ParserReqServiceImpl
- Adding SendKafkaControllerTest
- Rename tests in IntegrationTest
- Removing Currency
- Adding null checking for rating in ExportProductDtoList

viprikz committed:
Parser test

viprikz committed:
Merge remote-tracking branch 'origin/sprint-5' into sprint-5

kosch71 committed:
## ParserReqServiceImpl
- Adding properties for socket
- Adding rest template

kosch71 committed:
## ParserReqServiceImpl
- Deleting test

kosch71 committed:
## ParserReqServiceImpl
- Adding parsing for another wines
- Adding sparkling in ProductDTO
- Adding new Color and Sugar elements
- Renaming some methods

kosch71 committed:
## ParserReqServiceImpl
- Minor changes

grigorii-sum committed:
Refactoring for sonar

viprikz committed:
ParserConfiguration fix

viprikz committed:
Parser test

kosch71 committed:
## Minor changes
- Adding maven dependency
- Rename test name

kosch71 committed:
## Minor changes
- Deleting old dependence

kosch71 committed:
## Minor changes
- Renaming test names

viprikz committed:
Integration Test

viprikz committed:
Merge remote-tracking branch 'origin/sprint-5' into sprint-5

kosch71 committed:
## Minor changes
- Refactoring LentaWineParserServiceImplHelperTest

shinehlina commented on pom.xml:
```diff
@@ -29,10 +29,21 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+
```
Версию самого приложения)

shinehlina commented on pom.xml:
```diff
@@ -68,16 +82,52 @@
         <!-- Json -->
+
+        <dependency>
+            <groupId>com.wine.to.up</groupId>
```
Это не в блоке "Json"

shinehlina commented on pom.xml:
```diff
@@ -68,16 +82,52 @@
-            <groupId>com.wine.to.up</groupId>
-            <artifactId>parser-common-api</artifactId>
-            <version>${parser.common.api.version}</version>
+            <groupId>org.springframework.boot</groupId>
```
Вам недостаточно spring-boot-starter-test?

shinehlina commented on pom.xml:
```diff
@@ -68,16 +82,52 @@
+
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
```
Вам недостаточно spring-boot-starter-test?

kosch71 replied:
убрали

shinehlina commented on pom.xml:
```diff
@@ -68,16 +82,52 @@
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-webflux</artifactId>
```
Это зачем?

kosch71 replied:
убрали

shinehlina commented on pom.xml:
```diff
@@ -68,16 +82,52 @@
+
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
```
Этой зависимости нет в других? Что будет, если ее удалить?

kosch71 replied:
Убрал

shinehlina commented on pom.xml:
```diff
@@ -68,16 +82,52 @@
+
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
Эта зависимость точно есть уже

shinehlina commented on pom.xml:
```diff
@@ -68,16 +82,52 @@
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>3.3.3</version>
```
Эта точно тоже есть

shinehlina commented on src/main/java/com/wine/to/up/lenta/service/configuration/ParserConfiguration.java:
```diff
@@ -1,30 +1,37 @@
+    @Value("${site.rest.body}")
+    private String apiBody;
+
+    private HttpResponse<?> response;
```
Это что такое? Я не понимаю, где оно сеттится..

shinehlina commented on src/main/java/com/wine/to/up/lenta/service/controller/LentaStoreController.java:
```diff
@@ -1,30 +1,31 @@
-        list.add(parserReqService.getJsonList());
-        return list;
+    @GetMapping(name = "/parser", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> getParserResult() {
```
ResponseEntity<List>? Object вообще нехорошо)

kosch71 replied:
Теперь просто ResponseEntity

shinehlina commented on src/main/java/com/wine/to/up/lenta/service/cron/ExportProductDtoList.java:
```diff
@@ -16,20 +16,19 @@
 
     @Value("${string.for.cron}")
-    private String SHOP;
+    private String shopLink;
```
Не совсем понятно, почему в shopLink ставится string.for.cron. Что-то из этого лучше переименовать)

shinehlina commented on src/main/java/com/wine/to/up/lenta/service/parser/impl/ParserReqServiceImpl.java:
```diff
@@ -0,0 +1,186 @@
+
+
+
+    public ParserReqServiceImpl changeToLocal(){
```
Это что такое?

viprikz replied:
старый не используемый метод , убрали

shinehlina commented on src/main/resources/application.properties:
```diff
@@ -8,4 +8,6 @@ default.kafka.bootstrap-server=localhost:29092
-cron.expression=0 44 0 * * *
+cron.expression=0 19 18 * * *
+
+spring.elasticsearch.rest.read-timeout=60000s
```
Это для чего?

shinehlina commented on src/test/java/com/wine/to/up/lenta/service/parser/impl/ParserReqServiceImlpTest.java:
```diff
@@ -0,0 +1,50 @@
+
+@Validated
+@Slf4j
+public class ParserReqServiceImlpTest extends Mockito{
```
extends Mockito?
Validated?
Slf4j?

Зачем это

kosch71 replied:
Убрал

grigorii-sum committed:
Refactoring for review

kosch71 committed:
## Parsing changes
- Adding logging
- Adding csv controller
- Adding csv controller test
- Deleting non-usable dependencies in pom.xml

kosch71 committed:
## Minor changes
- Refactoring ParserReqServiceImlpTest
- Deleting property

kosch71 committed:
## Minor changes
- Deleting imports

kosch71 committed:
## Minor changes
- Deleting dependency

viprikz committed:
fix

viprikz committed:
Integration Test

shinehlina commented on src/main/java/com/wine/to/up/lenta/service/db/constans/Sugar.java:
```diff
@@ -13,25 +13,29 @@
-    public static UpdateProducts.Product.Sugar resolve(String sugar) {
-        return SUGAR_MAP.getOrDefault(sugar, Sugar.UNRECOGNIZED).productSugar;
+    public static ParserApi.Wine.Sugar resolve(String sugar) {
+        return SUGAR_MAP.getOrDefault(sugar, Sugar.DRY).productSugar;
```
Сделайте WARN, если приходится брать дефолт. (Хотя, может, стоит вообще ошибку кидать...)

kosch71 replied:
Исправил

shinehlina commented on src/main/java/com/wine/to/up/lenta/service/parser/impl/ParserReqServiceImpl.java:
```diff
@@ -0,0 +1,182 @@
+                    .build();
+
+            try {
+                response = client.send(request, HttpResponse.BodyHandlers.ofString());
```
Вы перезаписываете это поле, да? Это не очень хорошо, это все-таки класс - синглтон, который достуапен их нескольных потоков. Я не понимаю, почему это не локальная переменная

O-o-l-e-g replied:
В данном случае очень проблематично сделать это поле локально переменной. Тут либо вот такой вариант, либо очень сильно ругается Sonar, потому что получается очень много уязвимых для nullPointer мест

shinehlina replied:
Это просто идеологически неправильный подход... Может, вам стоит сделать это не со скоупом singleton... Вы же понимаете, почему это неправильно?

shinehlina commented on src/main/java/com/wine/to/up/lenta/service/parser/impl/ParserReqServiceImpl.java:
```diff
@@ -0,0 +1,182 @@
+                    } catch (IOException e) {
+                        log.error("Can't parse wine characteristics", e);
+                    } finally {
+                        wineList.add(getProperties(jsonString, document));
```
Если у вас произошла ошибка, то document null. Тогда вы передаете null в getProperties, а там стоит Objects.requiredNotNull. То есть вылетит нулпонтер

kosch71 replied:
Исправил

kosch71 committed:
## Review for PR from Elina
- Adding warn logs
- Adding try/catch for NullPointerException in ParserReqServiceImpl

kosch71 committed:
## Metrics
- Adding metrics for parsing and exporting to kafka

shinehlina reviewed (approved):
Ладно, сделайте это на след спринте. Только это обязательо нужно исправить
